tests/test_models/test_base_model.py

```diff
@@ -155,3 +162,228 @@ def test_model_with_no_outputs(self):
+TOTAL_TORQUE = 6.0
+AX = 7.0
+AY = 8.0
+
+
+@state_space_io(
+    states=("vx", "vy"), inputs=("throttle", "steer", "brake"), outputs=("ax", "ay"),
+)
+class TestVehicle(StateSpaceModel):
+    _submodel_names = ("powertrain", "tire")
+
+    @classmethod
+    def get_default_submodel_config(cls):
+        return {"powertrain": "HybridPowertrain", "tire": "MF72"}
+
+    def forward(self, states, inputs, mesh):
+        vx = self.get_state(states, "vx")
+        throttle = self.get_input(inputs, "throttle")
+
+        # Call powertrain model
+        powertrain_model = self.get_submodel("powertrain")
+        powertrain_inputs = powertrain_model.make_vector("inputs", demand=throttle)
+        powertrain_states = powertrain_model.make_vector(
+            "states", engine_speed=vx / 100.0
+        )
+
+        powertrain_return = powertrain_model.forward(
+            powertrain_states, powertrain_inputs, 0.0
+        )
+
+        # Call tire model
+        tire_model = self.get_submodel("tire")
+        tire_inputs = tire_model.make_const_vector("inputs", 0.3)
+        tire_return = tire_model.forward(tire_inputs)
+
+        submodel_outputs = self.combine_submodel_outputs(
+            powertrain=powertrain_return.outputs, tire=tire_return.outputs,
+        )
+
+        # Make some dummy outputs
+        outputs = self.make_vector("outputs", ax=AX, ay=AY, **submodel_outputs)
+        states_dot = self.make_vector("states_dot", vx_dot=0.1, vy_dot=0.2)
+        return self.make_state_space_model_return(
+            states_dot=states_dot, outputs=outputs
+        )
+
+
+@state_space_io(
+    states=("engine_speed",),
+    inputs=("demand",),
+    outputs=("total_power", "total_torque"),
+)
+class HybridPowertrain(StateSpaceModel):
+    _submodel_names = ("engine", "emotor")
+
+    @classmethod
+    def get_default_submodel_config(cls):
+        return {"engine": "SimpleEngine", "emotor": "SimpleEMotor"}
+
+    def forward(self, states, inputs, mesh):
+        demand = self.get_input(inputs, "demand")
+        engine_ratio = 0.8
+
+        # Assemble states_dot vector with dummy values
+        states_dot = self.make_vector("states_dot", engine_speed_dot=-25.0)
+
+        # Call engine
+        engine_model = self.get_submodel("engine")
+        engine_inputs = engine_model.make_vector("inputs", demand=demand * engine_ratio)
+        engine_return = engine_model.forward(engine_inputs)
+
+        # Call e-motor
+        emotor_model = self.get_submodel("emotor")
+        emotor_inputs = emotor_model.make_vector(
+            "inputs", demand=(1 - demand) * engine_ratio
+        )
+        emotor_return = emotor_model.forward(emotor_inputs)
+
+        # Combine all submodel outputs into a flattened dictionary
+        submodel_outputs = self.combine_submodel_outputs(
+            engine=engine_return.outputs, emotor=emotor_return.outputs,
+        )
+
+        # Use direct current model outputs and flattened submodel outputs to form the
```

Comment:
2) and then add the combined submodel outputs to the current model outputs (in addition to the direct outputs from the current model)